hass_nabucasa/voice.py

```diff
@@ -1234,15 +1234,15 @@ def _validate_token(self) -> bool:
 
     async def _update_token(self) -> None:
         """Update token details."""
-        resp = await cloud_api.async_voice_connection_details(self.cloud)
-        if resp.status != 200:
-            raise VoiceTokenError
+        try:
+            details = await self.cloud.voice_api.connection_details()
+        except VoiceApiError as err:
+            raise VoiceTokenError from err
 
-        data = await resp.json()
-        self._token = data["authorized_key"]
-        self._endpoint_stt = data["endpoint_stt"]
-        self._endpoint_tts = data["endpoint_tts"]
-        self._valid = utc_from_timestamp(float(data["valid"]))
+        self._token = details["authorized_key"]
+        self._endpoint_stt = details["endpoint_stt"]
+        self._endpoint_tts = details["endpoint_tts"]
+        self._valid = utc_from_timestamp(float(details["valid"]))
```

Comment:
Considering that `details` are of a specific type now, would it make sense to access properties like this?
```suggestion
        self._token = details.authorized_key
        self._endpoint_stt = details.endpoint_stt
        self._endpoint_tts = details.endpoint_tts
        self._valid = utc_from_timestamp(float(details.valid))
```

Reply:
This is not valid as the object is typedict.